src/lib.rs

```diff
@@ -5,3 +5,22 @@ pub mod config;
 pub mod install;
 pub mod pkg_config_gen;
 pub mod target;
+
+trait VersionExt {
+    /// build the main version string
+    fn main_version(&self) -> String;
+}
+
+impl VersionExt for semver::Version {
+    fn main_version(&self) -> String {
+        if self.major == 0 {
+            if self.minor == 0 {
+                format!("{}.{}.{}", self.major, self.minor, self.patch)
+            } else {
+                format!("{}.{}", self.major, self.minor)
+            }
+        } else {
+            format!("{}", self.major)
+        }
```

Comment:
```suggestion
        match (self.major, self.minor, self.patch) {
            (0, 0, patch) => format!("0.0.{patch}"),
            (0, minor, _) => format!("0.{minor}"),
            (major, _, _) => format!("{major}"),
        }
```

Reply:
It is a nice idea.

Reply:
Did you test that it works for you though? :)